llvm-external-projects/iree-dialects/lib/Dialect/LinalgExt/IR/LinalgExtOps.cpp

```diff
@@ -2201,8 +2202,28 @@ UnPackOp::getTiledImplementation(OpBuilder &builder,
                                           dimAndTileMapping[dim]));
 
       inputIndices.push_back(firstCoord.quotient);
-      inputSizes.push_back(
-          add(sub(lastCoord.quotient, firstCoord.quotient), oneAttr));
+
+      // Get the upper bound because it could be an extract_slice case. The
+      // sizes are determined by loop bound and step, where loop bound is the
+      // size of output shape. In incomplete tile cases, the input could have
+      // larger shape, it is safe to extend the boundary because they are
+      // pre-padded.
+      FailureOr<int64_t> cstSize = linalg::getConstantUpperBoundForIndex(
+          getValueOrCreateConstantIndexOp(builder, loc, sizes[dim]));
+      Optional<int64_t> cstInnerSize =
+          getConstantIntValue(dimAndTileMapping[dim]);
+      if (!failed(cstSize) && cstInnerSize &&
+          cstSize.value() % cstInnerSize.value() == 0) {
+        // Assume that offset[dim] must be 0 or a multiple of sizes[dim]. This
```

Comment:
Would make sense to assert this assumption?

Reply:
We can't assert this assumption because `offsets[dim]` is an IV. It is a Value, not a constant. :(